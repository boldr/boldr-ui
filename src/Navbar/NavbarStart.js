/* @flow */
import React from 'react';
import classNames from 'classnames';
import { createWrappedComponent } from '../util/boldrui';
import { StyleClasses } from '../theme/styleClasses';

export type Props = {
  tag?: string,
  className?: string,
};

const BASE_ELEMENT = StyleClasses.NAVBAR_START;

export function NavbarStart({ tag = 'div', ...props }: Props) {
  const className = classNames(BASE_ELEMENT, props.className);

  return React.createElement(tag, { ...props, className });
}

export default createWrappedComponent(NavbarStart);
