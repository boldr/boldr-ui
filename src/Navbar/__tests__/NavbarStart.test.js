import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { NavbarStart } from '../NavbarStart';

describe('<NavbarStart />', () => {
  it('should render a div with .boldr-navbar__start', () => {
    const component = shallow(<NavbarStart />);
    expect(component.contains(<div className="boldr-navbar__start" />)).toBe(true);
  });

  it('should render a p with .boldr-navbar__start', () => {
    const component = shallow(<NavbarStart tag="p" />);
    expect(component.contains(<p className="boldr-navbar__start" />)).toBe(true);
  });

  it('should render a div with boldr-navbar__start and custom classNames', () => {
    const component = shallow(<NavbarStart className="custom" />);
    expect(component.hasClass('boldr-navbar__start')).toBe(true);
    expect(component.hasClass('custom')).toBe(true);
  });
  it('should match the snapshot', () => {
    const wrapper = shallow(<NavbarStart />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
