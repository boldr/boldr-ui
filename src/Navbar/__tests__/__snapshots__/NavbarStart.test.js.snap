// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavbarStart /> should match the snapshot 1`] = `
<div
  className="boldr-navbar__start"
/>
`;
